packages/wrangler/src/pipelines/index.ts

```diff
@@ -0,0 +1,459 @@
+import yargs from "yargs";
+import { readConfig } from "../config";
+import { sleep } from "../deploy/deploy";
+import { FatalError } from "../errors";
+import { printWranglerBanner } from "../index";
+import { logger } from "../logger";
+import * as metrics from "../metrics";
+import { requireAuth } from "../user";
+import {
+	createPipeline,
+	deletePipeline,
+	generateR2ServiceToken,
+	getPipeline,
+	getR2Bucket,
+	listPipelines,
+	PartialExcept,
+	Pipeline,
+	PipelineUserConfig,
+	sha256,
+	updatePipeline,
+} from "./client";
+import type { CommonYargsArgv, CommonYargsOptions } from "../yargs-types";
+
+async function authorizeR2Bucket(
+	name: string,
+	accountId: string,
+	destination: Pipeline["destination"]
+) {
+	const bucket = destination.path.bucket;
+	if (!(await getR2Bucket(accountId, bucket))) {
+		logger.warn("Specified bucket doesn't exist:", bucket);
+	}
+
+	logger.log(`🌀 Authorizing R2 bucket "${bucket}"`);
+
+	const serviceToken = await generateR2ServiceToken(
+		`Service token for Pipeline ${name}`,
+		accountId,
+		bucket
+	);
+	const access_key_id = serviceToken.id;
+	const secret_access_key = sha256(serviceToken.value);
+
+	// wait for token to settle/propagate
+	await sleep(3000);
```

Comment:
I agree this isn't ideal, but the alternative seemed rougher.  The confirmation would require the use of the S3 API, which would be either:
1. much more complication, since we'd have to generate a signature ourselves: https://docs.aws.amazon.com/AmazonS3/latest/API/sigv4-auth-using-authorization-header.html
2. include a large package that implements the s3 standard, just to get endpoint verification.

Reply:
To address the delay in tests, I've added a delay bypass for tests.